pleskdistup/actions/spamassassin.py

```diff
@@ -31,3 +33,67 @@ def _post_action(self) -> action.ActionResult:
     def _revert_action(self) -> action.ActionResult:
         os.unlink(self.spamassasin_backup_path)
         return action.ActionResult()
+
+
+class HandleUpdatedSpamassassinConfig(action.ActiveAction):
+    # Make sure the trick is preformed before any call of 'systemctl daemon-reload'
+    # because we change spamassassin.service configuration in scope of this action.
+    def __init__(self):
+        self.name = "fix spamassassin configuration"
+
+    def _is_required(self) -> bool:
+        return packages.is_package_installed("psa-spamassassin")
+
+    def _prepare_action(self) -> action.ActionResult:
+        util.logged_check_call(["/usr/bin/systemctl", "stop", "spamassassin.service"])
+        util.logged_check_call(["/usr/bin/systemctl", "disable", "spamassassin.service"])
+        return action.ActionResult()
+
+    def _post_action(self) -> action.ActionResult:
+        util.logged_check_call(["/usr/sbin/plesk", "sbin", "spammng", "--enable"])
+        util.logged_check_call(["/usr/sbin/plesk", "sbin", "spammng", "--update", "--enable-server-configs", "--enable-user-configs"])
+
+        util.logged_check_call(["/usr/bin/systemctl", "daemon-reload"])
+        util.logged_check_call(["/usr/bin/systemctl", "enable", "spamassassin.service"])
+
+        if rpm.handle_rpmnew(SPAMASSASIN_CONFIG_PATH):
```

Comment:
What about Debian-based systems?

Reply:
This action will be just skipped on deb-based. So I believe we could just add TO-DO here if someone would like to use the action on deb-based.

Reply:
Do we have a check anywhere to skip it?

Reply:
We are looking for a .rpmnew file inside the function. However, we could add a check to make sure it is rpm-based system.